### PR TITLE
prevent infinite loops

### PR DESCRIPTION
**Pull request by ZenVoich**

### CODE REVIEW

File: src/Disburser/lib.mo [hunk 1]
```diff
@@ -78,15 +78,17 @@ module {
 
               switch (res) {
```
Comment (ZenVoich) [inline src/Disburser/lib.mo:79]: remove switch and add `i` to memo

File: src/Disburser/lib.mo [hunk 2]
```diff
@@ -78,15 +78,17 @@ module {
...
                 case (#Ok(blockIndex)) {};
-                case (#Err(#InsufficientFunds({ balance }))) {
+                case (#Err(#InsufficientFunds(_))) {
```
Comment (letmejustputthishere) [inline src/Disburser/lib.mo:81]: remove switch

File: src/Disburser/lib.mo [hunk 3]
```diff
@@ -78,15 +78,17 @@ module {
```
Comment (letmejustputthishere) [inline src/Disburser/lib.mo:78]: add "nonce" to memo. value between 0-100

File: src/Marketplace/lib.mo [hunk 4]
```diff
@@ -206,6 +204,9 @@ module {
       };
 
       if (response.e8s < settlement.price) {
+        if (response.e8s < 10_000) {
+          _tokenSettlement.delete(token);
+        };
```
Comment (letmejustputthishere) [inline src/Marketplace/lib.mo:209]: can be removed

**Review by letmejustputthishere — CHANGES_REQUESTED** [review]

**Review by letmejustputthishere — APPROVED** [review]

**Pull request merged**